pages/docs/tutorials/native/using-intellij-idea.md

```diff
@@ -1,50 +1,176 @@
+
+   The `src` directory contains Kotlin source files and resources. The file `main.kt` includes sample code that prints "Hello, Kotlin/Native!" using the [`prinln()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/println.html) function.
+
+2. Add the code to read the input. Use the [`readLine()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/read-line.html) function to read the input value and write it to the `name` variable.
+
+   <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+   ```kotlin
+   fun main() {
+       // Read the input value.
+       println("Hello, enter your name:")
+       val name = readLine()
+   }
+   ```
+
+   </div>
+
+3. Eliminate the white spaces and count the letters:
+   * Perform the null check of the input value with the [safe call operator `?.`](https://kotlinlang.org/docs/reference/null-safety.html#safe-calls).
+   * Use the [`replace()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/replace.html) function to eliminate the white spaces in the name.
+   * Use the scope function [`let`](https://kotlinlang.org/docs/reference/scope-functions.html#let) to execute function within the object context. 
+   * Use a [string template](https://kotlinlang.org/docs/reference/basic-types.html#string-templates) to insert your name length into the string by adding a dollar sign `$` and enclosing it in curly braces – `${it.length}`.
+     `it` is a default name of a [lambda parameter](https://kotlinlang.org/docs/reference/coding-conventions.html#lambda-parameters).
+
+   <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+   ```kotlin
+   fun main() {
+       // Read the input value.
+       println("Hello, enter your name:")
+       val name = readLine()
+       // Count letters in the name.
+       name?.replace(" ", "")?.let {
+           println("Your name contains ${it.length} letters")
+       }
+   }
+   ```
+
+   </div>
+
+4. Report about the null value using the [`error()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/error.html) function after the [Elvis operator `?:`](https://kotlinlang.org/docs/reference/null-safety.html#elvis-operator).
+
+   <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+   ```kotlin
+   fun main() {
+       // Read the input value.
+       println("Hello, enter your name:")
+       val name = readLine()
+       // Count letters in the name.
+       name?.replace(" ", "")?.let {
+           println("Your name contains ${it.length} letters")
+       } ?: error("Error while reading input from the terminal: the value can't be null.")
+   }
+   ```
+
+   </div>
+
+
+5. Save the changes and run the application.
+
+   IntelliJ IDEA opens the **Run** tab and shows the output.
+
+6. Enter your name and enjoy the result:
+
+   ![Application output]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-output-2.png') }})
```

Comment:
This may be a personal preference, but I think that using neutral example names is better than your real name. [Google agrees](https://developers.google.com/style/examples#names)

Reply:
Fixed. 
We should think about a common placeholder name for such cases.